### PR TITLE
LPS-173135 API functional test for APIFIlter in Playwright

### PR DESCRIPTION
**Pull request by magjed4289**
https://issues.liferay.com/browse/LPS-173135

### CODE REVIEW

**Comment by 4lejandrito** [comment]
@ccorreagg this is a rewrite of the `HeadlessBuilderResourceTest.testGetWithAPIFilter` in playwright to see what is best. Please review and provide your opinion.

File: modules/test/playwright/helpers/ApiHelpers.ts [hunk 1]
```diff
@@ -42,6 +45,22 @@ export class ApiHelpers {
...
+
+	async putResponse(url: string) {
+		const response = await this.page.request.put(url, {
+			headers: await this.getHeader(),
+		});
+
+		return response;
+	}
+
 	async post(url: string, data: DataObject) {
```
Comment (ccorreagg) [inline modules/test/playwright/helpers/ApiHelpers.ts:64]: Is this method needed or could `postResponse` be used?

File: modules/test/playwright/helpers/ApiBuilderHelper.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+	async getEndpointPage(endpointURI: string) {
+		return this.apiHelpers.getResponse(
+			`${this.apiHelpers.baseUrl}${endpointURI}`
+		);
+	}
+
+	async postApiApplication(data: DataObject) {
+		return this.apiHelpers.post(
+			`${this.apiHelpers.baseUrl}${this.basePath}/applications`,
```
Comment (ccorreagg) [inline modules/test/playwright/helpers/ApiBuilderHelper.ts:25]: It is a bit confusing having some paths ending with slash `/` (this.apiHelpers.baseUrl) and some are not (`this.basePath`). Could we unify them?

File: modules/test/playwright/helpers/ApiBuilderHelper.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+	async putResponse(uri: any) {
+		return this.apiHelpers.putResponse(
+			`${this.apiHelpers.baseUrl}${this.basePath}/${uri}`
+		);
+	}
+
+	async postAPIFilter(data: DataObject) {
+		return this.apiHelpers.postResponse(
+			`${this.apiHelpers.baseUrl}${this.basePath}/filters`,
```
Comment (ccorreagg) [inline modules/test/playwright/helpers/ApiBuilderHelper.ts:38]: I see we are repeating the same pattern here and in the `postApiApplication` method. Could we pass the data type (filter, application, etc) by argument?

File: modules/test/playwright/helpers/ObjectApiHelper.ts [hunk 4]
```diff
@@ -37,9 +38,9 @@ export class ObjectApiHelper {
 		};
 
 		return this.apiHelpers.postResponse(
-			`${this.apiHelpers.baseUrl}c/${
-				applicationName.toLowerCase() + 's'
-			}`,
+			`${this.apiHelpers.baseUrl}c/${formatPlural(
```
Comment (ccorreagg) [inline modules/test/playwright/helpers/ObjectApiHelper.ts:41]: I  just checked that we could get the path from the `restContextPath` field of the object definition:

```
curl -X GET -u test@liferay.com:test http://localhost:8080/o/object-admin/v1.0/object-definitions
```

```
{
    "accountEntryRestricted" : false,
    "accountEntryRestrictedObjectFieldName" : "",
    [...],
    "active" : true,
    "dateCreated" : "2024-01-22T10:11:23Z",
    "dateModified" : "2024-01-22T10:11:36Z",
    "defaultLanguageId" : "en_US",
    "enableCategorization" : true,
    "enableComments" : false,
    "enableLocalization" : false,
    "enableObjectEntryDraft" : false,
    "enableObjectEntryHistory" : false,
    "externalReferenceCode" : "6790f1a2-1e3a-6503-be37-899169f21487",
    "id" : 114129,
    "label" : {
      "en_US" : "Test"
    },
    "modifiable" : true,
    "name" : "Test",
    "objectActions" : [ ],
    "objectFields" : [ {
       [...],
    } ],
    "objectLayouts" : [ ],
    "objectRelationships" : [ ],
    "objectValidationRules" : [ ],
    "objectViews" : [ ],
    "panelCategoryKey" : "",
    "parameterRequired" : false,
    "pluralLabel" : {
      "en_US" : "PluralTest"
    },
    "portlet" : true,
    "restContextPath" : "/o/c/tests",
    "scope" : "company",
    "status" : {
      "code" : 0,
      "label" : "approved",
      "label_i18n" : "Approved"
    },
    "system" : false,
    "titleObjectFieldName" : "id"
  }
```

Comment (ccorreagg) [reply modules/test/playwright/helpers/ObjectApiHelper.ts:41]: That could apply to our integration tests too

**Review by ccorreagg — COMMENTED** [review]
Please, see my comments, I think it could be a good idea to have functional tests for API Builder, but I am not sure if it provides us something new compared to the integration tests

File: modules/test/playwright/helpers/ApiHelpers.ts (hunk 1 above)
Comment (magjed4289) [reply modules/test/playwright/helpers/ApiHelpers.ts:64]: in some methods we get data directly from json response, other response elements are not needed, I'll reuse the code of postResponse to call it

**Comment by magjed4289** [comment]
resent [here](https://github.com/liferay-headless/liferay-portal/pull/1876)

**Pull request closed without merging**